packages/core/src/common/resource.ts

```diff
@@ -341,8 +347,24 @@ export class UntitledResourceResolver implements ResourceResolver {
     }
 
     async createUntitledResource(content?: string, extension?: string, uri?: URI): Promise<UntitledResource> {
-        return new UntitledResource(this.resources, uri ? uri : new URI().withScheme(UNTITLED_SCHEME).withPath(`/Untitled-${untitledResourceSequenceIndex++}${extension ?? ''}`),
-            content);
+        if (!uri) {
+            uri = this.createUntitledURI(extension);
+        }
+        return new UntitledResource(this.resources, uri, content);
+    }
+
+    createUntitledURI(extension?: string, parent?: URI): URI {
+        let counter: number = 1; // vscode is started from 1
```

Comment:
Minor: We can let TypeScript infer the type.
```suggestion
        let counter = 1; // vscode starts at 1
```